src/notes/notes.controller.ts

```diff
@@ -0,0 +1,98 @@
+import {
+  Body,
+  Controller,
+  Delete,
+  Get,
+  HttpStatus,
+  Param,
+  Post,
+  Put,
+  Query,
+  Res,
+  UseGuards,
+} from '@nestjs/common';
+import { Response } from 'express';
+
+import { NotesService } from 'notes/notes.service';
+import {
+  NameDto,
+  UpdateNoteDto,
+  CreateNoteDto,
+  IdDto,
+  FilterPaginationDto,
+} from 'dto';
+import {
+  BASE_NOTES_URLS,
+  CONTENT_TYPE,
+  CONTENT_TYPE_HTML,
+  ROLES,
+  URL_PREFIX,
+} from 'common';
+import { JwtAuthGuard, RoleGuard } from 'auth/guard';
+import { Roles } from 'auth/role.decorator';
+
+@Controller(URL_PREFIX)
+export class NotesController {
+  constructor(private readonly notesService: NotesService) {}
+
+  @Get(BASE_NOTES_URLS.GREETINGS)
+  getName(@Param() { name }: NameDto, @Res() response: Response): Response {
+    const responseMessage = this.notesService.getHelloWithName(name);
+
+    return response
+      .setHeader(CONTENT_TYPE, CONTENT_TYPE_HTML)
+      .status(HttpStatus.OK)
+      .send(responseMessage);
+  }
+
+  @Roles(ROLES.USER)
+  @UseGuards(JwtAuthGuard, RoleGuard)
+  @Get(BASE_NOTES_URLS.NOTES)
+  async getNotes(
+    @Query() parameters: FilterPaginationDto,
+    @Res() response: Response,
+  ): Promise<Response> {
+    const notes = await this.notesService.getNotes(parameters);
+
+    return response.send(notes);
+  }
+
+  @Roles(ROLES.USER)
+  @UseGuards(JwtAuthGuard, RoleGuard)
+  @Post(BASE_NOTES_URLS.NOTES)
+  async insertNote(
+    @Body() note: CreateNoteDto,
+    @Res() response: Response,
+  ): Promise<Response> {
+    const createdNote = await this.notesService.createNote(note);
+
+    return response.status(HttpStatus.OK).send(createdNote);
+  }
+
+  @Roles(ROLES.USER)
+  @UseGuards(JwtAuthGuard, RoleGuard)
+  @Put(BASE_NOTES_URLS.NOTES_BY_ID)
```

Comment:
should we use put or patch for updating?

Reply:
I can update title and content hole entity, because put